### ISSUE TITLE
IndexError with merged cells that extend beyond right edge of non-merged data

### ISSUE THREAD

**Issue by ochawkeye** [python-excel/xlrd#129]
Based on the commentary on #113 I do not believe this is related, but very well could be...

I'm experiencing some inconsistent results with merged cells that are wider than any other data in a worksheet and it's possible I'm just not handling these cells correctly.

```python
import xlrd

wb = xlrd.open_workbook('xlrd_IndexError1.xlsx')
#wb = xlrd.open_workbook('xlrd_IndexError2.xlsx')

for sheet in wb.sheets():
    print 'Sheet:', sheet.name
    for row in range(sheet.nrows):
        print 'Row:', row
        for col in range(sheet.ncols):
            print '\tCol:', col, sheet.cell(row, col).value
    print
```
Two pretty similar situations with two pretty different results.
[First example](https://github.com/ochawkeye/xlrd/blob/IndexError/tests/xlrd_IndexError1.xlsx):
![image](https://cloud.githubusercontent.com/assets/2996205/6218258/1a406a9e-b5e2-11e4-9309-02c03aff75ea.png)
![image](https://cloud.githubusercontent.com/assets/2996205/6218259/227c496c-b5e2-11e4-80f4-8fadb2efd5d9.png)

 Two sheets; Sheet 1 has merged cell that goes only as far right as the right-most data (being C2).  Sheet 2 has merged cell that extends further to the right than C2.  Sheet 1 is read fine, but error when trying to read contents of merged cell on Sheet 2.
```
Sheet: Sheet1
Row: 0
        Col: 0 Title
        Col: 1
        Col: 2
Row: 1
        Col: 0 A2
        Col: 1 B2
        Col: 2 C2
Row: 2
        Col: 0 A3
        Col: 1 B3C3
        Col: 2

Sheet: Sheet2
Row: 0
        Col: 0 Title
        Col: 1
        Col: 2
        Col: 3
Row: 1
        Col: 0 A2
        Col: 1 B2
        Col: 2 C2
        Col: 3
Traceback (most recent call last):
  File "xlrd_issue.py", line 11, in <module>
    print '\tCol:', col, sheet.cell(row, col).value
  File "C:\Python27\lib\site-packages\xlrd\sheet.py", line 399, in cell
    self._cell_types[rowx][colx],
IndexError: array index out of range
```
[Second example](https://github.com/ochawkeye/xlrd/blob/IndexError/tests/xlrd_IndexError2.xlsx): 
![image](https://cloud.githubusercontent.com/assets/2996205/6218270/3836c660-b5e2-11e4-9041-df480533ee84.png)
Similar tables setup, but both are on Sheet 1 now.  Fail to read row 2 of first table.
```
Sheet: Sheet1
Row: 0
        Col: 0 Title
        Col: 1
        Col: 2
        Col: 3
Row: 1
        Col: 0 A2
        Col: 1 B2
        Col: 2 C2
        Col: 3
Traceback (most recent call last):
  File "xlrd_issue.py", line 11, in <module>
    print '\tCol:', col, sheet.cell(row, col).value
  File "C:\Python27\lib\site-packages\xlrd\sheet.py", line 399, in cell
    self._cell_types[rowx][colx],
IndexError: array index out of range
```

Is there a different strategy to use to handle these or is this actually an issue?

**Comment by ochawkeye** [comment]
[One more example](https://github.com/ochawkeye/xlrd/blob/IndexError/tests/xlrd_IndexError3.xlsx)
![image](https://cloud.githubusercontent.com/assets/2996205/6218242/ee5ee306-b5e1-11e4-821b-c93522cbf1c3.png)

This sheet has all table cells properly contained in the "envelope" of the data in the rest of the sheet, but has an empty merged cell that falls outside.

```
Sheet: Sheet1
Row: 0
        Col: 0 Title
        Col: 1
        Col: 2
        Col: 3
        Col: 4
        Col: 5
Row: 1
        Col: 0 A2
        Col: 1 B2
        Col: 2 C2
        Col: 3
Traceback (most recent call last):
  File "xlrd_issue.py", line 12, in <module>
    print '\tCol:', col, sheet.cell(row, col).value
  File "C:\Python27\lib\site-packages\xlrd\sheet.py", line 399, in cell
    self._cell_types[rowx][colx],
IndexError: array index out of range
```

**Comment by cjw296** [comment]
Fixed in 1.0.1.
